### PR TITLE
Get portfolio balance 

### PR DESCRIPTION
**Pull request by AlexeyBarabash**
Resolves https://github.com/brave/brave-browser/issues/18098

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [ ] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [ ] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:
1. Add some accounts 
2. Ensure the portfolio balance is updated

### CODE REVIEW

File: browser/ui/android/strings/android_brave_strings.grd [hunk 1]
```diff
@@ -579,6 +579,9 @@ month...
       <message name="IDS_BRAVE_UI_BACKUP_WALLET_DESC" desc="A notification message description about backing up the wallet">
         Please backup your Brave wallet.
       </message>
+      <message name="IDS_BRAVE_WALLET_PORTFOLIO_BALANCE_UPDATING">
```
Comment (SergeyZhukovsky) [inline browser/ui/android/strings/android_brave_strings.grd:582]: add `translateable="false"` argument as we don't want to translate it

File: android/java/org/chromium/chrome/browser/crypto_wallet/fragments/PortfolioFragment.java [hunk 2]
```diff
@@ -170,4 +226,116 @@ private void setUpCoinList(View view) {
...
+    // TODO(AlexeyBarabash): this is duplicated over other wallet UI classes
+    public AssetRatioController getAssetRatioController() {
+        return mAssetRatioController;
+    }
+    private void InitAssetRatioController() {
+        if (mAssetRatioController != null) {
+            return;
+        }
+
+        mAssetRatioController =
```
Comment (SergeyZhukovsky) [inline android/java/org/chromium/chrome/browser/crypto_wallet/fragments/PortfolioFragment.java:246]: we want to do all mojo connections in the parent Activity, `BraveWalletActivity` in that case. And get all interfaces from there.

File: android/java/org/chromium/chrome/browser/crypto_wallet/fragments/PortfolioFragment.java [hunk 3]
```diff
@@ -56,6 +76,7 @@ private EthJsonRpcController getEthJsonRpcController() {
     public void onCreate(@Nullable Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setHasOptionsMenu(true);
+        getEthJsonRpcController().addObserver(this);
```
Comment (SergeyZhukovsky) [inline android/java/org/chromium/chrome/browser/crypto_wallet/fragments/PortfolioFragment.java:79]: Why do we need that observer? Only we control when we change network. And we send that event in `public void onItemSelected(AdapterView<?> parent, View view, int position, long id) {`. We can just hook update balance there

File: browser/ui/android/strings/android_brave_strings.grd (hunk 1 above)
Comment (AlexeyBarabash) [reply browser/ui/android/strings/android_brave_strings.grd]: thanks, fixed with https://github.com/brave/brave-core/pull/10174/commits/2c83ac338c7c317997db171e4bd4ad21b79bcc1f

File: android/java/org/chromium/chrome/browser/crypto_wallet/fragments/PortfolioFragment.java (hunk 2 above)
Comment (AlexeyBarabash) [reply android/java/org/chromium/chrome/browser/crypto_wallet/fragments/PortfolioFragment.java]: thanks, fixed with https://github.com/brave/brave-core/pull/10174/commits/6cb21e371d38518475bf20df2a47d0fe7edcef5b

File: android/java/org/chromium/chrome/browser/crypto_wallet/fragments/PortfolioFragment.java [hunk 4]
```diff
@@ -84,22 +100,43 @@ public boolean onTouch(View v, MotionEvent event) {
...
+            }
+        });
 
         // Creating adapter for spinner
         ArrayAdapter<String> dataAdapter = new ArrayAdapter<String>(getActivity(),
                 android.R.layout.simple_spinner_item, Utils.getNetworksList(getActivity()));
         dataAdapter.setDropDownViewResource(android.R.layout.simple_spinner_dropdown_item);
         mSpinner.setAdapter(dataAdapter);
 
+        RadioGroup radioGroup = view.findViewById(R.id.portfolio_duration_radio_group);
```
Comment (SergeyZhukovsky) [inline android/java/org/chromium/chrome/browser/crypto_wallet/fragments/PortfolioFragment.java:127]: unused var?

Comment (AlexeyBarabash) [reply android/java/org/chromium/chrome/browser/crypto_wallet/fragments/PortfolioFragment.java:127]: yes, indeed

Comment (AlexeyBarabash) [reply android/java/org/chromium/chrome/browser/crypto_wallet/fragments/PortfolioFragment.java]: Fixed in https://github.com/brave/brave-core/pull/10174/commits/4c0d233d43308b0dfd339f73d69ac9fff4b4d458

**Review by SergeyZhukovsky — APPROVED** [review]
++

**Comment by AlexeyBarabash** [comment]
Squashed commits, waiting for CI

**Pull request merged**